pkg/webui/console/components/device-data-form/index.js

```diff
@@ -36,44 +36,82 @@ import PropTypes from '../../../lib/prop-types'
 import m from './messages'
 import validationSchema from './validation-schema'
 
-@bind
 class DeviceDataForm extends Component {
   constructor(props) {
     super(props)
 
-    const { initialValues } = this.props
+    const {
+      initialValues: { supports_join, ids, root_keys = {} },
+    } = this.props
+
+    const external_js = !Boolean(Object.keys(root_keys).length)
     let otaa = true
-    if (initialValues.ids) {
-      otaa = Boolean(initialValues.supports_join)
+    if (ids) {
```

Comment:
```suggestion
    if (update) {
```
Can you use the update prop instead?